File: Chandra Kiran/6-10-2022/twosumii.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
    for (i in numbers) {
        const res = bsearch(numbers, (target - numbers[i]),i);
        if (res) {
            return [parseInt(i)+1, res+1];
        }
   }
};

var bsearch = function (nums, target,i) {
    low = 0;
    high = nums.length - 1;
    while (low <= high) {
        mid = Math.ceil((low + high) / 2);
        if (nums[mid] == target) {
            if (mid != i)
                return mid;
            return mid + 1;
        }
        else if (nums[mid] > target) {
            high = mid - 1;
        }
        else if (nums[mid] < target) {
            low = mid + 1;
        }
    }
    return 0;
}

console.log(twoSum([1, 3,4,4], 8));

File: Jhanavi/6-10-2022/Two_Sum_II.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l,r = 0 , len(numbers)-1
        
        while l<r :
            cSum = numbers[l] + numbers[r]
            
            if cSum > target:
                r-=1
            elif cSum < target :
                l+=1
            else :
                return [l+1 , r+1]
        return []    
